packages/amplify-cli-core/src/state-manager/stateManager.ts

```diff
@@ -0,0 +1,145 @@
+
+  getTeamProviderInfo = (projectPath?: string, options?: GetOptions<$TSTeamProviderInfo>): $TSTeamProviderInfo => {
+    const filePath = pathManager.getTeamProviderInfoFilePath(projectPath);
+    const mergedOptions = {
+      throwIfNotExist: true,
+      ...options,
+    };
+
+    return this.getData<$TSTeamProviderInfo>(filePath, mergedOptions);
+  };
+
+  isLocalEnvInfoExists = (projectPath?: string): boolean => fs.existsSync(pathManager.getLocalEnvFilePath(projectPath));
+
+  getLocalEnvInfo = (projectPath?: string, options?: GetOptions<$TSAny>): $TSAny => {
+    const filePath = pathManager.getLocalEnvFilePath(projectPath);
+    const mergedOptions = {
+      throwIfNotExist: true,
+      ...options,
+    };
+
+    return this.getData<$TSAny>(filePath, mergedOptions);
+  };
+
+  getLocalAWSInfo = (projectPath?: string, options?: GetOptions<$TSAny>): $TSAny => {
+    const filePath = pathManager.getLocalAWSInfoFilePath(projectPath);
+    const mergedOptions = {
+      throwIfNotExist: true,
+      ...options,
+    };
+
+    return this.getData<$TSAny>(filePath, mergedOptions);
+  };
+
+  getProjectConfig = (projectPath?: string, options?: GetOptions<$TSAny>): $TSAny => {
+    const filePath = pathManager.getProjectConfigFilePath(projectPath);
+    const mergedOptions = {
+      throwIfNotExist: true,
+      ...options,
+    };
+
+    return this.getData<$TSAny>(filePath, mergedOptions);
+  };
+
+  getBackendConfig = (projectPath?: string, options?: GetOptions<$TSAny>): $TSAny => {
+    const filePath = pathManager.getBackendConfigFilePath(projectPath);
+    const mergedOptions = {
+      throwIfNotExist: true,
+      ...options,
+    };
+
+    return this.getData<$TSAny>(filePath, mergedOptions);
+  };
+
+  setLocalEnvInfo = (projectPath: string | undefined, localEnvInfo: $TSAny): void => {
+    const filePath = pathManager.getLocalEnvFilePath(projectPath);
+
+    JSONUtilities.writeJson(filePath, localEnvInfo);
+  };
+
+  setLocalAWSInfo = (projectPath: string | undefined, localAWSInfo: $TSAny): void => {
+    const filePath = pathManager.getLocalAWSInfoFilePath(projectPath);
+
+    JSONUtilities.writeJson(filePath, localAWSInfo);
+  };
+
+  setProjectConfig = (projectPath: string | undefined, projectConfig: $TSAny): void => {
+    const filePath = pathManager.getProjectConfigFilePath(projectPath);
+
+    JSONUtilities.writeJson(filePath, projectConfig);
+  };
+
+  setTeamProviderInfo = (projectPath: string | undefined, teamProviderInfo: $TSTeamProviderInfo): void => {
+    const filePath = pathManager.getTeamProviderInfoFilePath(projectPath);
+
+    JSONUtilities.writeJson(filePath, teamProviderInfo);
+  };
+
+  setBackendConfig = (projectPath: string | undefined, backendConfig: $TSAny): void => {
+    const filePath = pathManager.getBackendConfigFilePath(projectPath);
+
+    JSONUtilities.writeJson(filePath, backendConfig);
+  };
+
+  setMeta = (projectPath: string | undefined, meta: $TSMeta): void => {
```

Comment:
```suggestion
  setMeta = (meta: $TSMeta, projectPath?: string): void => {
```

Reply:
As we discussed the reason for the suggestion offline, I would not change it now as `projectPath` should be either mandatory when supplied and behavior should not depend it as the path of the project should be handled internally within `StateManager` later.